pandas/tseries/index.py

```diff
@@ -308,6 +311,9 @@ def __new__(cls, data=None,
                                                      dayfirst=dayfirst,
                                                      yearfirst=yearfirst)
 
+        if data.dtype in (_INT64_DTYPE, _NS_DTYPE) and copy:
```

Comment:
I think you should do the copy at the very end, you might be double copying here. now its tricky to figure that out.